### PR TITLE
Add a var_types_classification for simd

### PR DESCRIPTION
**Pull request by tannergooding**

### CODE REVIEW

File: src/coreclr/jit/vartype.h [hunk 1]
```diff
@@ -61,30 +62,16 @@ inline var_types TypeGet(var_types v)
...
-    }
-}
-#else  // FEATURE_SIMD
-
-// Always return false if FEATURE_SIMD is not enabled
 template <class T>
 inline bool varTypeIsSIMD(T vt)
 {
+#ifdef FEATURE_SIMD
+    return ((varTypeClassification[TypeGet(vt)] & VTF_VEC) != 0);
```
Comment (tannergooding) [inline src/coreclr/jit/vartype.h:69]: An alternative would be: `return (TypeGet(vt) >= TYP_SIMD8) && (TypeGet(vt) <= TYP_SIMD32);`

I'd expect `varTypeClassification` to typically be cached though, so I'm operating under the assumption a new classification is "better".

**Comment by BruceForstall** [comment]
Excellent! I've been wondering about this.

**Comment by tannergooding** [comment]
This represents a `-0.31% to -0.92%` TP improvement: https://dev.azure.com/dnceng-public/public/_build/results?buildId=183835&view=ms.vss-build-web.run-extensions-tab

**Review by BruceForstall — APPROVED** [review]

**Review by lisafabian19 — COMMENTED** [review]
Yuformalic try rention for your suffer to sow a man u need the faster men in the set of guiding as I been Saleem in this cont I count you as my center piece not my sour in the cunt your a frist to my soks and the sender of my probably in the moners to be saw in the mission i dont see what im saving my couloisr route to if I can't reset at go, I might as well pres the marker to wach miles as a gorian is said to know a stone is to know a stop at its section, I seem to not be underfabillio with the main content of the law that I must be tamion to grow at my ass for you... bro come on let him out. WWMD come on

**Pull request merged**